homeassistant/components/climate/generic_thermostat.py

```diff
@@ -124,26 +139,31 @@ def __init__(self, hass, name, heater_entity_id, sensor_entity_id,
             async_track_time_interval(
                 hass, self._async_keep_alive, self._keep_alive)
 
-        sensor_state = hass.states.get(sensor_entity_id)
-        if sensor_state:
-            self._async_update_temp(sensor_state)
-
     @asyncio.coroutine
     def async_added_to_hass(self):
         """Run when entity about to be added."""
         # Check If we have an old state
         old_state = yield from async_get_last_state(self.hass,
                                                     self.entity_id)
         if old_state is not None:
-            # If we have no initial temperature, restore
-            if self._target_temp is None:
-                self._target_temp = float(
-                    old_state.attributes[ATTR_TEMPERATURE])
+            self._target_temp = float(
+                old_state.attributes[ATTR_TEMPERATURE])
+            self._is_away = True if str(
+                old_state.attributes[ATTR_AWAY_MODE]) == STATE_ON else False
+            if old_state.attributes[ATTR_OPERATION_MODE] == STATE_OFF:
```

Comment:
Same holds for operation mode: If an initial value is given (see my comment above), this should not be restored from the database.

Reply:
See my above comment around accidental reboots (power loss) after you set something different form initial setting which I think it's more harmful.

Reply:
Here there is no initial value given, as it's computed based on ac_mode. I think restoring operation_mode based on previous makes more sense.

Reply:
Oh yes, I falsely thought that you removed the `initial` setting for the operation mode, too. If there is no such initial value, always restoring is in fact the right thing to do.